### PR TITLE
javadoc - Adding default constructor and javadoc for :ethereum:api

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
javadoc - Adding default constructor and javadoc for :ethereum:api (~200 files)

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [x] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLConfiguration.java [hunk 1]
```diff
@@ -26,8 +26,11 @@
 
 import com.google.common.base.MoreObjects;
 
+/** The type Graph ql configuration. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLConfiguration.java:29]:
```suggestion
/** The type GraphQL configuration. */
```

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLConfiguration.java:29]: prob worth a find and replace

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcApis.java [hunk 2]
```diff
@@ -19,30 +19,50 @@
...
   ETH,
+  /** Debug rpc apis. */
   DEBUG,
+  /** Miner rpc apis. */
   MINER,
+  /** Net rpc apis. */
   NET,
+  /** Perm rpc apis. */
   PERM,
+  /** Web 3 rpc apis. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcApis.java:34]:
```suggestion
  /** Web3 rpc apis. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java [hunk 3]
```diff
@@ -17,186 +17,366 @@
...
   TRACE_TRANSACTION("trace_transaction"),
+  /** Tx pool besu statistics rpc method. */
   TX_POOL_BESU_STATISTICS("txpool_besuStatistics"),
+  /** Tx pool besu transactions rpc method. */
   TX_POOL_BESU_TRANSACTIONS("txpool_besuTransactions"),
+  /** Tx pool besu pending transactions rpc method. */
   TX_POOL_BESU_PENDING_TRANSACTIONS("txpool_besuPendingTransactions"),
+  /** Web 3 client version rpc method. */
   WEB3_CLIENT_VERSION("web3_clientVersion"),
+  /** Web 3 sha 3 rpc method. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:366]:
```suggestion
  /** Web3 sha3 rpc method. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java [hunk 4]
```diff
@@ -17,186 +17,366 @@
...
   TRACE_REPLAY_BLOCK_TRANSACTIONS("trace_replayBlockTransactions"),
+  /** Trace transaction rpc method. */
   TRACE_TRANSACTION("trace_transaction"),
+  /** Tx pool besu statistics rpc method. */
   TX_POOL_BESU_STATISTICS("txpool_besuStatistics"),
+  /** Tx pool besu transactions rpc method. */
   TX_POOL_BESU_TRANSACTIONS("txpool_besuTransactions"),
+  /** Tx pool besu pending transactions rpc method. */
   TX_POOL_BESU_PENDING_TRANSACTIONS("txpool_besuPendingTransactions"),
+  /** Web 3 client version rpc method. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:364]:
```suggestion
  /** Web3 client version rpc method. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java [hunk 5]
```diff
@@ -14,15 +14,29 @@
  */
 package org.hyperledger.besu.ethereum.api.jsonrpc.authentication;
 
+/** The enum Jwt algorithm. */
 public enum JwtAlgorithm {
+  /** Rs 256 jwt algorithm. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java:19]:
```suggestion
  /** RS256 jwt algorithm. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java [hunk 6]
```diff
@@ -14,15 +14,29 @@
...
   RS256,
+  /** Rs 384 jwt algorithm. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java:21]:
```suggestion
  /** RS384 jwt algorithm. */
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java [hunk 7]
```diff
@@ -14,15 +14,29 @@
...
   RS384,
+  /** Rs 512 jwt algorithm. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java:23]:
```suggestion
  /** RS512 jwt algorithm. */
```

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/JwtAlgorithm.java:23]: prob same for all of these

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetCode.java [hunk 8]
```diff
@@ -25,12 +25,23 @@
 
 import org.apache.tuweni.bytes.Bytes;
 
+/** The type Eth get code. */
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetCode.java:28]: maybe it is worth changing these? idk
```suggestion
/** eth_getCode method. */
```

**Review by macfarla — COMMENTED** [review]
request global find/replace "[Gg]raph ql" to "GraphQL" 
otherwise looks pretty ok

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/ApiConfiguration.java [hunk 9]
```diff
@@ -18,64 +18,126 @@
 
 import org.immutables.value.Value;
 
+/** The type Api configuration. */
```
Comment (siladu) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/ApiConfiguration.java:21]: Wondering if this comment is configurable with your tool? It's turning the class name into a sentence which is a potentially a little confusing, e.g. there's no "Api" type here

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthChainId.java [hunk 10]
```diff
@@ -23,10 +23,16 @@
 import java.math.BigInteger;
 import java.util.Optional;
 
+/** The type Eth chain id. */
 public class EthChainId implements JsonRpcMethod {
 
   private final Optional<BigInteger> chainId;
 
+  /**
+   * Instantiates a new Eth chain id.
```
Comment (siladu) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthChainId.java:32]: same comment with this, probably anything that refers to the classname I guess

**Review by siladu — COMMENTED** [review]
Is it possible to prevent the tool from turning the classname into a "sentence"?

**Comment by usmansaleem** [comment]
Will recreate with smaller PRs

**Pull request closed without merging**